astro

```diff
@@ -376,10 +376,9 @@ EOF
 	esac
 	debug "charset: $charset"
 
-	case $4 in
-		*".gmi"|"") typesetgmi ;;
-		*.*) cat ;;
-		*) typesetgmi ;;
+	case ${meta#[0-9][0-9]" "} in
```

Comment:
Hmmm, this `${#}` is specific to bash, we cannot use this, it breaks POSIX standard.

Whats is wrong with meta to need this?

I think line 380 is enough to be 
`*"text/gemini"* | "") typesetgmi ;;`

Am I missing something?

Reply:
hm - according to https://pubs.opengroup.org/onlinepubs/9699919799/ this behaviour is defined for POSIX shells and `shellcheck` does not complain about it being undefined in POSIX.

`meta` contains the status code as well (e.g. `10 text/gemini`), `""` would not match.

Reply:
nevermind - i finally get what you where talking about :facepalm: